dap_prinz_green_jobs/pipeline/green_measures/occupations/soc_map.py

```diff
@@ -0,0 +1,378 @@
+"""
+A class to map inputted job titles to their most likely SOC 2020 4-digit codes.
+
+Usage:
+
+soc_mapper = SOCMapper()
+soc_mapper.load()
+matches = soc_mapper.get_soc(job_titles=["data scientist", "Assistant nurse", "Senior financial consultant - London"])
+>>> [('2425', 'data scientist'), ('6141', 'assistant nurse'), ('3534', 'financial consultant')]
+
+"""
+from collections import Counter, defaultdict
+import os
+from typing import List, Union
+
+from sentence_transformers import SentenceTransformer
+from sklearn.metrics.pairwise import cosine_similarity
+import pandas as pd
+from tqdm import tqdm
+import numpy as np
+
+from dap_prinz_green_jobs.getters.occupation_getters import load_job_title_soc
+from dap_prinz_green_jobs.getters.data_getters import (
+    save_to_s3,
+    load_s3_data,
+    load_json_dict,
+)
+from dap_prinz_green_jobs import BUCKET_NAME, logger, config, PROJECT_DIR
+
+
+def chunks(orig_list, n_chunks):
+    for i in range(0, len(orig_list), n_chunks):
+        yield orig_list[i : i + n_chunks]
```

Comment:
some version of this function already exists in `dap_prinz_green_jobs/pipeline/green_measures/skills/skill_measures_utils.py` (lines 6 - 19) - feels like we should pull chunking lists/dicts out into its own i.e. `dap_prinz_green_jobs/utils/processing.py` file instead of task specific files